### PR TITLE
feat: new proof request loading screen

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

**Improved loading experience and streamlined testing**

* **New Loading Animations**: You'll see a smoother loading experience when establishing connections and handling proof requests.
* **Centralized Mock Data**: Test data for OOB, Connection, Offer, and Proof is now neatly organized in `react-hooks.ts`, making tests easier to manage.
* **Test & Snapshot Updates**: Unit tests and snapshots have been refreshed to reflect these changes.

This summary focuses on the key user-facing improvement (loading animations) and the behind-the-scenes benefit of centralized mock data, while also acknowledging the necessary test updates. It uses positive language and avoids technical jargon that might not be familiar to all reviewers. 

# Related Issues

n/a

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj
Comment (bryce-mcmath) [inline packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj:1]: Should we not commit this file?

File: packages/legacy/core/App/hooks/notifications.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
 import { useBasicMessages, useCredentialByState, useProofByState } from '@credo-ts/react-hooks'
 import { ProofCustomMetadata, ProofMetadata } from '@hyperledger/aries-bifold-verifier'
 import { useEffect, useState } from 'react'
+// import { log } from 'console'
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/hooks/notifications.ts]: should probably remove

File: packages/legacy/core/App/components/animated/RecordLoading.tsx [hunk 2]
```diff
@@ -1,55 +1,72 @@
...
+            }}
+          />
+          <View
+            style={{
+              marginLeft: -22,
+              marginTop: 20,
+              backgroundColor: ColorPallet.grayscale.lightGrey,
+              height: 45,
+              width: 45,
+              borderRadius: 5,
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/components/animated/RecordLoading.tsx]: These width / margin numbers seem somewhat arbitrary, could they be explained or named?

File: packages/legacy/core/App/components/animated/RecordLoading.tsx [hunk 3]
```diff
@@ -1,55 +1,72 @@
...
+            }}
+          />
+          <View style={{ marginLeft: 15, marginTop: 15, marginBottom: 15 }}>
+            <View style={[myStyle.rectangle, { width: 240, height: 20 }]} />
+            <View style={[myStyle.rectangle, { width: 145, height: 25 }]} />
+            <View style={[myStyle.rectangle, { width: 75, height: 20, marginTop: 20 }]} />
+            <View style={[myStyle.rectangle, { width: 210, height: 25 }]} />
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/components/animated/RecordLoading.tsx]: Same here

**Review by bryce-mcmath — APPROVED** [review]

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]
Sorry, meant to request changes but accidentally approved

**Review by bryce-mcmath — DISMISSED** [review]

**Comment by bryce-mcmath** [comment]
The xcode file is still included but that isn't a big deal it's just an EOF

**Comment by jleach** [comment]
> The xcode file is still included but that isn't a big deal it's just an EOF

It was just a EOF but I fixed it anyway.

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**